Content.Server/_Sunrise/Anomaly/Systems/AnomalyAutoInjectorSystem.cs

```diff
@@ -0,0 +1,188 @@
+    [Dependency] private readonly IGameTiming _timing = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<AnomalyAutoInjectorComponent, AfterInteractEvent>(OnAfterInteract);
+    }
+
+
+
+    // ShowPopup wrapper removed per review
+
+    private bool IsValidTargetForInjection(EntityUid target, EntityUid injector, AnomalyAutoInjectorComponent comp, [NotNullWhen(false)] out string? popup)
+    {
+        popup = null;
+
+        if (!HasComp<MobStateComponent>(target))
+        {
+            popup = comp.PopupNotApplicable;
+            return false;
+        }
+
+        if (!HasComp<HumanoidAppearanceComponent>(target))
+        {
+            if (!HasComp<UsedAnomalyAutoInjectorComponent>(injector))
+                popup = comp.PopupNotApplicable;
+            else
+                popup = comp.PopupNothingToInject;
+            return false;
+        }
+
+        if (HasComp<UsedAnomalyAutoInjectorComponent>(injector))
+        {
+            popup = comp.PopupNothingToInject;
+            return false;
+        }
+
+        if (HasComp<PendingAnomalyInfectionComponent>(target))
+        {
+            popup = comp.PopupPending;
+            return false;
+        }
+
+        if (HasComp<InnerBodyAnomalyComponent>(target))
+        {
+            popup = comp.PopupInfected;
+            return false;
+        }
+
+        return true;
+    }
+
+    private void OnAfterInteract(EntityUid uid, AnomalyAutoInjectorComponent comp, AfterInteractEvent args)
+    {
+        if (args.Handled || !args.CanReach || args.Target is not { } target)
+            return;
+
+        if (!IsValidTargetForInjection(target, uid, comp, out var popup))
+        {
+            if (popup != null)
+                _popup.PopupEntity(popup, target, args.User);
+            return;
+        }
+
+        // Проверка конфигурации до любых побочных эффектов.
+        if (comp.AnomalyTrapProtos.Count == 0)
+        {
+            Logger.Error($"AnomalyAutoInjector {ToPrettyString(uid)} has empty AnomalyTrapProtos list");
+            args.Handled = true;
+            return;
+        }
+
+        EnsureComp<UsedAnomalyAutoInjectorComponent>(uid);
+        args.Handled = true;
+        if (Exists(uid))
+            _audio.PlayPvs(comp.HypospraySound, uid);
+
+        _statusEffects.TryAddStatusEffectDuration(target, comp.RainbowEffect, TimeSpan.FromSeconds(comp.RainbowDuration));
+        if (TryComp<SeeingRainbowsWeakStatusEffectComponent>(target, out var rainbowComp))
+        {
+            rainbowComp.Intensity = comp.RainbowEffectIntensity;
+            Dirty(target, rainbowComp);
+        }
+
+        var pending = EnsureComp<PendingAnomalyInfectionComponent>(target);
+        pending.EndAt = _timing.CurTime + TimeSpan.FromSeconds(comp.AnomalyDelay);
+        pending.CellularDamage = comp.CellularDamage;
+        pending.SelectedAnomalyTrapProtoId = _random.Pick(comp.AnomalyTrapProtos);
```

Comment:
```suggestion
        EnsureComp<UsedAnomalyAutoInjectorComponent>(uid); // Снова же вопрос, почему бы не использовать bool в уже наложенном компоненте
        args.Handled = true;
  
        _audio.PlayPvs(comp.HypospraySound, uid);

        _statusEffects.TryAddStatusEffectDuration(target, comp.RainbowEffect, TimeSpan.FromSeconds(comp.RainbowDuration));
   
        if (!TryComp<SeeingRainbowsWeakStatusEffectComponent>(target, out var rainbowComp))
            return;
        
        rainbowComp.Intensity = comp.RainbowEffectIntensity;
        Dirty(target, rainbowComp);
    
        var pending = EnsureComp<PendingAnomalyInfectionComponent>(target);
        pending.EndAt = _timing.CurTime + TimeSpan.FromSeconds(comp.AnomalyDelay);
        pending.CellularDamage = comp.CellularDamage;
        pending.SelectedAnomalyTrapProtoId = _random.Pick(comp.AnomalyTrapProtos);
```
Смысл проверять Exists нет, так как ты подписался на ивент, и он тупо не сработает если uid пропадет или тип того